agcaballero/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -73,15 +76,29 @@ public String toString() {
     /**
      * @return a Comment as a HTML div with proper formatting to be displayed
      */
-    private String htmlFormat() {
+    private String htmlFormat(String languageCode) {
       // gets time in local time zone (default: US ET)
       LocalDateTime localTime =
           new LocalDateTime(timestamp.getTime(), DateTimeZone.forID("US/Eastern"));
       DateTimeFormatter formatter = DateTimeFormat.forPattern("h:mm a M/dd/yy");
 
       return "<div class='comment-div'>"
           + "<p class='date'>" + formatter.print(localTime) + "</p>"
-          + "<p><b>" + name + " (" + email + "):</b> <br><br>" + text + "</p></div>";
+          + "<p><b>" + name + " (" + email + "):</b></p>"
+          + "<p class='comment-text' lang=" + languageCode + ">" + text + "</p></div>";
+    }
+
+    /**
+     * Translate the text to the language of a given language code
```

Comment:
Might also be worth pointing out that this synchronously makes a network call, so could be slow (or should this method be made asynchronous?)

Reply:
@ykjchen I think clarifying in the comment it could be slow is sufficient to me. Do you think it should be made asynchronous? I haven't done that with a Java method before.

Reply:
In the interest of time, I'd go with the comment as the MVP... but for future reference perhaps a CompletableFuture would be appropriate for tasks like this:
https://www.baeldung.com/java-asynchronous-programming#completableFuture

i.e. the method would return CompletableFuture<String> rather than String.

Reply:
Thank you for the info! If I have time later this week, I might try this out.

Reply:
Done